pallets/runtime/tests/src/settlement_test.rs

```diff
@@ -3066,3 +3083,371 @@ fn multiple_custodian_settlement() {
+        EncryptedAssetIdWrapper::from(mercat_account_tx.pub_account.enc_asset_id.encode()),
+    )
+}
+
+pub fn create_account_and_mint_token(
+    owner: Public,
+    owner_did: IdentityId,
+    total_supply: u128,
+    token_name: Vec<u8>,
+    tx_id: u32,
+    seed: [u8; 32],
+) -> u32 {
+    let funding_round_name: FundingRoundName = b"round1".into();
+
+    let token = SecurityToken {
+        name: token_name.clone().into(),
+        owner_did,
+        total_supply: total_supply,
+        divisible: false,
+        asset_type: AssetType::default(),
+        primary_issuance_agent: Some(owner_did),
+        ..Default::default()
+    };
+    let ticker = Ticker::try_from(token.name.as_slice()).unwrap();
+    let identifiers = vec![(IdentifierType::default(), b"undefined".into())];
+
+    assert_ok!(ConfidentialAsset::create_confidential_asset(
+        Origin::signed(owner),
+        token.name.clone(),
+        ticker,
+        true,
+        token.asset_type.clone(),
+        identifiers.clone(),
+        Some(funding_round_name.clone()),
+    ));
+
+    // In the initial call, the total_supply must be zero.
+    assert_eq!(Asset::token_details(ticker).total_supply, Zero::zero());
+
+    // ---------------- prepare for minting the asset
+
+    let valid_asset_ids: Vec<AssetId> = ConfidentialAsset::confidential_tickers();
+
+    let (scrt_account, mercat_account_tx) = gen_account(tx_id, seed, &token_name, valid_asset_ids);
+
+    ConfidentialAsset::validate_mercat_account(Origin::signed(owner), mercat_account_tx.clone())
+        .unwrap();
+
+    // ------------- Computations that will happen in owner's Wallet ----------
+    let amount: u32 = token.total_supply.try_into().unwrap(); // mercat amounts are 32 bit integers.
+    let mut new_seed = seed;
+    new_seed[0] += 1;
+    let mut rng = StdRng::from_seed(new_seed);
+    let issuer_account = Account {
+        scrt: scrt_account,
+        pblc: mercat_account_tx.pub_account.clone(),
+    };
+
+    let tx_id = tx_id + 1;
+    let initialized_asset_tx = AssetIssuer {}
+        .initialize_asset_transaction(tx_id, &issuer_account, &[], amount, &mut rng)
+        .unwrap();
+
+    // Wallet submits the transaction to the chain for verification.
+    ConfidentialAsset::mint_confidential_asset(
+        Origin::signed(owner),
+        ticker,
+        amount.into(), // convert to u128
+        initialized_asset_tx,
+    )
+    .unwrap();
+
+    // ------------------------- Ensuring that the asset details are set correctly
+
+    // A correct entry is added.
+    assert_eq!(
+        Asset::asset_ownership_relation(token.owner_did, ticker),
+        AssetOwnershipRelation::AssetOwned
+    );
+    assert_eq!(Asset::funding_round(ticker), funding_round_name.clone());
+
+    // Ticker is added to the list of confidential tokens.
+    assert_eq!(
+        ConfidentialAsset::confidential_tickers().last(),
+        Some(&AssetId {
+            id: *ticker.as_bytes()
+        })
```

Comment:
I am not sure if we can do this. The AssetId is defined in cryptography lib, while the Ticker is defined in Polymesh. Adding such a function would make Polymesh a dependency of cryptography lib, which is not desirable.

Reply:
It's just the primitives crate that the crypto crate would need to depend on though? Not everything else. Can you elaborate on why that is a problem?

Reply:
We would like to keep the cryptography library as standalone as possible. So that it can potentially be used in other projects. Therefore, we are picky about what dependencies we add. For example, in this case, it would tightly couple the library to our version of subtrate primitives.